File: QuizAppServer/src/database/QuizDao.java
package database;

import common.Question;

import java.sql.SQLException;
import java.util.ArrayList;

public interface QuizDao {
    public ArrayList<Question> fetchQuestions() throws SQLException;
    public int addQuestion(Question questions[]) throws SQLException;
    public int removeQuestion(Question question);
    public int updateQuestion(Question question);
    public Question fetchSpecific(int id) throws SQLException;
}

File: QuizAppServer/src/database/QuizDao.kt
package database

import common.Question
import java.sql.SQLException
import java.util.*

interface QuizDao {
    @Throws(SQLException::class)
    fun fetchQuestions(): ArrayList<Question?>

    @Throws(SQLException::class)
    fun addQuestion(questions: Array<Question>): Int
    fun removeQuestion(question: Question?): Int
    fun updateQuestion(question: Question?): Int

    @Throws(SQLException::class)
    fun fetchSpecific(id: Int): Question
}